src/main.ts

```diff
@@ -67,10 +67,12 @@ const basicAuth: BasicAuth | null =
       }
     : null;
 
+// yargs pass type definition including undifined and when default option is undifined, yargs catchs error.
+// therefore, use as assertion in run function
 run(
-  argv["base-url"],
-  argv.username,
-  argv.password,
+  argv["base-url"] as string,
+  argv.username as string,
+  argv.password as string,
```

Comment:
Type assertion with as is the easy way to override type,  but I think it is not appropriate in case that there is another way.

I recommend you use User-Defined Type Guards.
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

If the value is undefined, it could be error and bugs. Then, it should log the error.